src/Centreon/Infrastructure/CentreonLegacyDB/Mapping/ClassMetadata.php

```diff
@@ -0,0 +1,223 @@
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ * PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, see <http://www.gnu.org/licenses>.
+ *
+ * Linking this program statically or dynamically with other modules is making a
+ * combined work based on this program. Thus, the terms and conditions of the GNU
+ * General Public License cover the whole combination.
+ *
+ * As a special exception, the copyright holders of this program give Centreon
+ * permission to link this program with independent modules to produce an executable,
+ * regardless of the license terms of these independent modules, and to copy and
+ * distribute the resulting executable under terms of Centreon choice, provided that
+ * Centreon also meet, for each linked independent module, the terms  and conditions
+ * of the license of that module. An independent module is a module which is not
+ * derived from this program. If you modify this program, you may extend this
+ * exception to your version of the program, but you are not obliged to do so. If you
+ * do not wish to do so, delete this exception statement from your version.
+ *
+ * For more information : contact@centreon.com
+ *
+ *
+ */
+
+namespace Centreon\Infrastructure\CentreonLegacyDB\Mapping;
+
+use PDO;
+
+class ClassMetadata
+{
+    const COLUMN = 'column';
+    const TYPE = 'type';
+    const FORMATTER = 'formatter';
+
+    /**
+     * Table name of entity
+     *
+     * @var string
+     */
+    protected $tableName;
+
+    /**
+     * List of properties of entity vs columns in DB table
+     *
+     * @var array
+     */
+    protected $columns;
+
+    /**
+     * Name of property that is the primary key
+     *
+     * @var string
+     */
+    protected $primaryKey;
+
+    /**
+     * 
+     * @param string $name
+     * @return \self
+     */
+    public function setTableName($name): self
+    {
+        $this->tableName = $name;
+
+        return $this;
+    }
+
+    /**
+     * Get table name of entity
+     *
+     * @return string
+     */
+    public function getTableName(): ?string
+    {
+        return $this->tableName;
+    }
+
+    /**
+     * Add information about the property
+     *
+     * @param string $property      name of the property in the Entity class
+     * @param string $columnName    name of the column in DB
+     * @param int $dataType         type of data use PDO::PARAM_*
+     * @param callable $dataFormatter
+     * @param bool $primaryKey      is it PK
+     * @return \self
+     */
+    public function add(
+        string $property,
+        string $columnName,
+        int $dataType = PDO::PARAM_STR,
+        callable $dataFormatter = null,
+        $primaryKey = false
+    ): self
```

Comment:
```suggestion
    ): self {
```
remove warning from phpcs